### ISSUE TITLE
Kombu does correct support multi-threading sqla broker

### ISSUE THREAD

**Issue by CrazyLionHeart** [celery/kombu#1395]
In case running Celery with `sqla+postgresql+psycopg2` broker with prefork model I facing with problem - all broker connection is died.

In logs we found error - `This session in "prepared" state; no further SQL can be emmited within this transaction`.

Short research show sqla broker does not use thread-local session object - https://docs.sqlalchemy.org/en/14/orm/contextual.html

**Comment by auvipy** [comment]
yeah the broker still need work on SQLA 1.4 release. we just added basic compatibility

**Comment by ebertti** [comment]
FYI for anyone here who's specifically on PostgreSQL (not tied to Oracle/MySQL/etc. via SQLAlchemy's dialect abstraction): I built [`kombu-pgmq`](https://github.com/ebertti/kombu-pgmq), a Kombu transport for the [PGMQ](https://github.com/pgmq/pgmq) Postgres extension that talks to Postgres directly (via `psycopg` 3 or the official `pgmq` client), not through a shared SQLAlchemy `Session`. It sidesteps this exact class of bug — no ORM session object crossing thread/prefork boundaries, since each call either owns its own connection or pulls one from a `psycopg_pool.ConnectionPool`.

Not a fix for the `sqla` transport itself (still very much worth fixing on its own merits), just flagging it as an option if PostgreSQL-only is enough for your use case. I'm the author.
